### DISCUSSION TITLE
Can this extension work with Remote - SSH?

### DISCUSSION THREAD

**Discussion by mddrill** [scalameta/metals#2628] [Q&A]
I am trying to use this extension with a project that I connect to using the [Visual Studio Code Remote - SSH](https://code.visualstudio.com/docs/remote/ssh) extension. When I open my project the "Starting Metals Server..." notification shows in the bottom right but hangs forever. I can't seem to find a way to get any more detailed log output.

I am able to start the metals server with projects that are not on a remote machine. Is there a way that I can get this extension working with remote ssh?

**Comment by tgodzik** [comment]
Last time I tried it worked, is there anything in the remote machines's `.metals/metals.log` file? It might be having problems downloading the server itself, is the machine behind a proxy etc. ?

**Reply by mddrill** [reply]
It doesn't seem like it's even getting to the step where `.metals/` files are created. I can't find any output anywhere, so I have nothing to go off of. Here's a picture:

<img width="958" alt="Screen Shot 2021-03-22 at 3 46 49 PM" src="https://user-images.githubusercontent.com/20046178/112049300-d21b7e00-8b25-11eb-9781-5d7896bd1bd4.png">

I wouldn't be surprised if the machine is behind a proxy, but I don't know for sure. Where could I look for log files to see if it's getting stuck?

**Reply by tgodzik** [reply]
The only place we would be getting some information at this point would be in the Output, which seems to be empty.

It seems to be hanging on finding java home:
```
        const javaHome = await getJavaHome(
          workspace.getConfiguration("metals").get("javaHome")
        );
```

Could you try specifying the path to java home in the container?

**Reply by mddrill** [reply]
Specifying the java home got me past that step, but now I'm seeing

```
INFO  sbt bloopInstall exit: 0
INFO  time: ran 'sbt bloopInstall' in 12s
INFO  Attempting to connect to the build server...
ERROR Failed to connect with build server, no functionality will work.
java.lang.RuntimeException: The server did not start, got FailedToOpenBspConnection
	at bloop.launcher.LauncherMain.failPromise$1(Launcher.scala:95)
	at bloop.launcher.LauncherMain.runLauncher(Launcher.scala:122)
	at scala.meta.internal.metals.BloopServers$$anon$1.run(BloopServers.scala:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I tried specifying the sbt script location in the workspace settings, but that didn't make a difference.

**Reply by tgodzik** [reply]
That one looks like if the Bloop build server was not downloaded. The alternative would be to run `metals.generate-bsp-config` command and connect to sbt. Or you can also try to run Bloop inside the container via [coursier](https://get-coursier.io/):

`cs install bloop` followed up by `bloop about`

**Reply by Enhowww** [reply]
As a newbie, I tried updating the JDK to JDK17, then restarting, and the problem was fixed. My ubuntu version is 22.04, running on a virtual machine and using ssh to remotely connect to vscode for windows
